src/rest/impl/operations.service.ts

```diff
@@ -100,152 +101,93 @@ export default class OperationsServiceImpl extends OperationsService {
-        case OPERATION_SAP_ITEM_GROUP_SYNC:
-          return this.readSapImportItemGroups(activeSapData.data);
-        case OPERATION_SAP_CONTRACT_SYNC:
-          return this.readSapImportContracts(sapDatas);
-      }
-
-    } catch (e) {
-      this.logger.error(`Failed to parse SAP import file ${e}`);
     }
-
-    return null;
   }
 
   /**
-   * Reads business partners from sap and fills related task queue with data from file
-   * 
-   * @param {Object} sap SAP data object 
+   * Reads business partners from sap and fills related task queue with data
    */
-  private async readSapImportBusinessPartners(sap: any) {
-    if (!sap.BusinessPartners) {
-      this.logger.error("Failed to read SAP business parterns");
-      return;
-    }
+  private async readSapImportBusinessPartners() {
+    try {
+      const sapBusinessPartnersService = SapServiceFactory.getBusinessPartnersService();
+      const businessPartners = await sapBusinessPartnersService.listBusinessPartners();
+      const operationReport = await models.createOperationReport("SAP_CONTACT_SYNC");
+
+      businessPartners.forEach((businessPartner: any) => {
+        tasks.enqueueSapContactUpdate(operationReport.id, businessPartner);
+      });
 
-    const businessPartners = sap.BusinessPartners.BusinessPartners;
-    if (!businessPartners) {
-      this.logger.error("Failed to read SAP business parterns list");
+      return operationReport;
+    } catch (e) {
+      this.logger.error(`Failed to read SAP item groups. error: ${e}`);
       return;
     }
-
-    const operationReport = await models.createOperationReport("SAP_CONTACT_SYNC");
-
-    businessPartners.forEach((businessPartner: any) => {
-      tasks.enqueueSapContactUpdate(operationReport.id, businessPartner);
-    });
-
-    return operationReport;
   }
 
   /**
-   * Reads delivery places from sap and fills related task queue with data from file
-   * 
-   * @param {Object} sap SAP data object 
+   * Reads delivery places from sap and fills related task queue with data
    */
-  private async readSapImportDeliveryPlaces(sap: SAPExportRoot) {
-    if (!sap.DeliveryPlaces) {
-      this.logger.error("Failed to read SAP item groups");
-      return;
-    }
+  private async readSapImportDeliveryPlaces() {
+    try {
+      const sapDeliveryPlacesService = SapServiceFactory.getDeliveryPlacesService();
+      const deliveryPlaces = await sapDeliveryPlacesService.listDeliveryPlaces();
+      const operationReport = await models.createOperationReport(OPERATION_SAP_DELIVERY_PLACE_SYNC);
 
-    const deliveryPlaces = sap.DeliveryPlaces.DeliveryPlaces;
-    if (!deliveryPlaces) {
-      this.logger.error("Failed to read SAP delivery places list");
+      deliveryPlaces.forEach((deliveryPlace) => {
+        tasks.enqueueSapDeliveryPlaceUpdate(operationReport.id, deliveryPlace);
+      });
+  
+      return operationReport;
+    } catch (e) {
+      this.logger.error(`Failed to read SAP item groups. error: ${e}`);
```

Comment:
Shoudn't this throw an error?

Reply:
Same here, undefined answer from this promise will throw 500 from function that calls this